packages/cli/internal/pkg/cli/spec/context.go

```diff
@@ -1,19 +1,46 @@
 package spec
 
+import "fmt"
+
 const DefaultMaxVCpus = 256
+const DefaultFSProvisionedThroughput = 0
 
+type FSConfig struct {
+	FSProvisionedThroughput int `yaml:"provisionedThroughput"`
+}
+type Filesystem struct {
+	FSType        string   `yaml:"fsType"`
+	Configuration FSConfig `yaml:"configuration,omitempty"`
+}
 type Engine struct {
-	Type   string `yaml:"type"`
-	Engine string `yaml:"engine"`
+	Type       string     `yaml:"type"`
+	Engine     string     `yaml:"engine"`
+	Filesystem Filesystem `yaml:"filesystem,omitempty"`
 }
-
 type Context struct {
 	InstanceTypes        []string `yaml:"instanceTypes,omitempty"`
 	RequestSpotInstances bool     `yaml:"requestSpotInstances,omitempty"`
 	MaxVCpus             int      `yaml:"maxVCpus,omitempty"`
 	Engines              []Engine `yaml:"engines"`
 }
 
+func (filesystem *Filesystem) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	type defValFilesystem Filesystem
+	defaults := defValFilesystem{Configuration: FSConfig{FSProvisionedThroughput: DefaultFSProvisionedThroughput}}
+	if err := unmarshal(&defaults); err != nil {
+		return err
+	}
+
+	*filesystem = Filesystem(defaults)
+	switch filesystem.FSType {
+	case "S3", "EFS", "":
+		return nil
+	default:
+		return fmt.Errorf("filesystem %s is invalid. Options are `S3` and `EFS`", filesystem.FSType)
```

Comment:
nit: can you wrap the string variable in single quotes? It's generally easy to read/understand

```
fmt.Errorf("filesystem '%s' is invalid. Options are `S3` and `EFS`", filesystem.FSType)
```